src/pages/Components/Section/ChangePW.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react';
+import { ReactComponent as Prev } from 'assets/svg/prev-icon.svg';
+import warning from 'assets/svg/warning.svg';
+import style from './SearchId.module.scss';
+import ModalForPW from './ModalForPW';
+
+export default function ChangePW() {
+  const pattern = /^.*(?=^.{2,16}$)(?=.*\d)(?=.*[a-zA-Z])(?=.*[!@#$%^&+=]).*$/;
+  const [pw, setPw] = useState<string>('');
+  const [pwch, setPwch] = useState<string>('');
+  const [Valid, setValid] = useState<string>('initial');
+  const [isEmpty, setIsEmpty] = useState(true);
+  const [isSuccess, setIsSuccess] = useState(false);
+  const checkEmpty = () => {
+    if (pw !== '' && pwch !== '') setIsEmpty(false);
+    else if (pw === '' || pwch === '') setIsEmpty(true);
+  };
+  const setPassword = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPw(e.target.value);
+    checkEmpty();
+  };
+  const setPasswordCheck = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setPwch(e.target.value);
+    checkEmpty();
+  };
+  const checking = (e: React.MouseEvent<HTMLButtonElement>) => {
+    if (pw === '' || pwch === '') {
+      e.preventDefault();
+      setIsEmpty(true);
+    } else if (pw !== '' && pwch !== '') {
+      if (pw === pwch && !pattern.test(pw)) {
+        e.preventDefault();
+        setValid('patternError');
+      } else if (pw !== pwch) {
+        e.preventDefault();
+        setValid('different');
+      } else {
+        e.preventDefault();
+        setValid('initial');
+        setIsSuccess(true);
+      }
+    }
+  };
```

Comment:
로직이 알아보기 너무 힘드네요 😭 아래와 같이 수정하면 충분해보여요 (혹시 다르나 확인 바랍니다)
그리고 preventdefault 에 대해 더 학습하시면 좋을것 같아요

```suggestion
  const checking = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault();
    
    if (pw === '' || pwch === '') {  
      setIsEmpty(true);
      return;
    } 
    
    if(pw !== pwch){
      setValid('different');
      return;
    }
    
    if(!pattern.test(pw)){
      setValid('patternError');
      return;
    }
    
    setValid('initial');
    setIsSuccess(true);
  };
```

Reply:
넵 참고하겠습니다
추가적으로 부족한 부분도 더 공부하겠습니다